### PR TITLE
Updating ContainerService event grid event schema to support new events

### PR DESCRIPTION
**Pull request by tristanang**
# Data Plane API - Pull Request
AKS is adding support for additional events via Event Grid. Thus, we need to update the swagger definitions.

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [x] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document: https://microsoft-my.sharepoint.com/:w:/p/kaarthis/EXCnUiin1oxHmQ8FYm2z5z0BSmfxrgG4p2xGUfNyLnK-qg?e=WnN7sW
* Previous API Spec Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

**Comment by tristanang** [comment]
@JoshLove-msft for visibility

**Review by Brandon411 — APPROVED** [review]

File: specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json [hunk 1]
```diff
@@ -27,6 +27,58 @@
...
+    },
+    "ContainerServiceNodePoolRollingProgressEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.ContainerService.NodePoolRollingProgress event",
+      "type": "object",
+      "properties": {
+        "nodePoolName": {
+          "description": "The name of the node pool in the ManagedCluster resource",
+          "type": "string"
+        },
+        "progressState": {
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json:70]: Please make this an enum. Here is an example of how to do this - https://github.com/Azure/azure-rest-api-specs/blob/main/specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json#L497-L508

File: specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json [hunk 2]
```diff
@@ -27,6 +27,58 @@
...
+    },
+    "ContainerServiceClusterOutOfSupportEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.ContainerService.ClusterOutOfSupport event",
+      "type": "object",
+      "properties": {
+        "kubernetesVersion": {
+          "description": "The Kubernetes version of the ManagedCluster resource",
+          "type": "string"
+        },
+        "outOfSupport": {
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json:48]: nit: should we call it `isOutOfSupport`?

Comment (tristanang) [reply specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json]: sounds clearer

File: specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json (hunk 1 above)
Comment (tristanang) [reply specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json:70]: thanks, think I updated it correctly

File: specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json [hunk 3]
```diff
@@ -27,6 +27,67 @@
...
+        "progressState": {
+          "description": "Indicates the progress of the node pool rolling process. Possible values are: \"Started\", \"Succeeded\", \"Failed\"",
+          "type": "string",
+          "enum": [
+            "Started",
+            "Succeeded",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "progressState",
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json:79]: This name is probably too generic - what about something like nodePoolProgressState?

Comment (tristanang) [reply specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json]: Updated it to rollingProgressState, let me know if you feel it's still too generic

File: specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json [hunk 4]
```diff
@@ -27,6 +27,104 @@
...
+      "x-ms-examples": {
+        "clusterSupportEndedCloudEventsSchema": {
+          "$ref": "./examples/cloud-events-schema/cluster_support_ended.json"
+        },
+        "clusterSupportEndedEventGridSchema": {
+          "$ref": "./examples/event-grid-schema/cluster_support_ended.json"
+        }
+      }
+    },
+    "ContainerServiceClusterSupportEndingEventData": {
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json:58]: Can we add a base model that both definitions reference named `ContainerServiceClusterSupportBaseProperties`?

File: specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json [hunk 5]
```diff
@@ -27,6 +27,104 @@
...
+      "x-ms-examples": {
+        "nodePoolRollingStartedCloudEventsSchema": {
+          "$ref": "./examples/cloud-events-schema/node_pool_rolling_started.json"
+        },
+        "nodePoolRollingStartedEventGridSchema": {
+          "$ref": "./examples/event-grid-schema/node_pool_rolling_started.json"
+        }
+      }
+    },
+    "ContainerServiceNodePoolRollingSucceededEventData": {
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.ContainerService/stable/2018-01-01/ContainerService.json:94]: Can we add a base model that each definition references called `ContainerServiceNodePoolRollingBaseProperties`?

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**